pkg/kubectl/cmd/top_pod.go

```diff
@@ -129,58 +129,15 @@ func (o TopPodOptions) RunTopPod() error {
 		}
 	}
 	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, selector)
+	if err != nil {
+		return err
+	}
 	// TODO: Refactor this once Heapster becomes the API server.
```

Comment:
This TODO refers to the error above - as stated, this will be refactored after implementing generic apiserver in Heapster.

The workaround was described here: [https://github.com/kubernetes/kubernetes/pull/31510#discussion_r76408471](https://github.com/kubernetes/kubernetes/pull/31510#discussion_r76408471)